cdc/owner.go

```diff
@@ -1626,3 +1684,28 @@ func (o *Owner) startCaptureWatcher(ctx context.Context) {
 		}
 	}()
 }
+
+// handle the StaleChangeFeed
+// By setting the AdminJob type to AdminStop and the Error code to indicate that the changefeed is stagnant
+func (o *Owner) handleStaleChangeFeed(ctx context.Context, staleChangeFeedID []model.ChangeFeedID, minGcSafePoint, gcSafePoint uint64) error {
+	for _, id := range staleChangeFeedID {
+		message := cerror.ErrStartTsBeforeGC.GenWithStackByArgs(minGcSafePoint, gcSafePoint).Error()
+		log.Info("handle staleChangeFeed ", zap.String("changefeed", id), zap.String("Error message", message))
+
+		runningError := &model.RunningError{
+			Addr:    util.CaptureAddrFromCtx(ctx),
+			Code:    string(cerror.ErrStartTsBeforeGC.RFCCode()), // changfeed is stagnant
```

Comment:
`ErrStartTsBeforeGC` might not be the most appropriate error code. We would like the user to get more useful information.